qiskit/transpiler/passes/utils/native_cr_direction.py

```diff
@@ -0,0 +1,137 @@
+from qiskit.transpiler.exceptions import TranspilerError
+
+from qiskit.circuit import QuantumRegister
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.circuit.library.standard_gates import RYGate, HGate, CXGate, ECRGate, RZXGate
+
+
+class NativeCRGateDirection(TransformationPass):
+    """Modify asymmetric gates to match the hardware coupling direction.
+    This pass makes use of the following identities::
+                             ┌───┐┌───┐┌───┐
+        q_0: ──■──      q_0: ┤ H ├┤ X ├┤ H ├
+             ┌─┴─┐  =        ├───┤└─┬─┘├───┤
+        q_1: ┤ X ├      q_1: ┤ H ├──■──┤ H ├
+             └───┘           └───┘     └───┘
+             ┌──────┐          ┌───────────┐┌──────┐┌───┐
+        q_0: ┤0     ├     q_0: ┤ RY(-pi/2) ├┤1     ├┤ H ├
+             │  ECR │  =       └┬──────────┤│  ECR │├───┤
+        q_1: ┤1     ├     q_1: ─┤ RY(pi/2) ├┤0     ├┤ H ├
+             └──────┘           └──────────┘└──────┘└───┘
+             ┌─────────┐            ┌───┐┌─────────┐┌───┐
+        q_0: ┤0        ├       q_0: ┤ H ├┤1        ├┤ H ├
+             │  Rzx(θ) │  =         ├───┤│  Rzx(θ) │├───┤
+        q_1: ┤1        ├       q_1: ┤ H ├┤0        ├┤ H ├
+             └─────────┘            └───┘└─────────┘└───┘
+    """
+
+    def __init__(self, backend):
+        """GateDirection pass.
+        Args:
+            backend (BaseBackend): The hardware backend that is used.
+        """
+        super().__init__()
+        self.backend = backend
+
+        # Create the replacement dag and associated register.
+        self._cx_dag = DAGCircuit()
+        qr = QuantumRegister(2)
+        self._cx_dag.add_qreg(qr)
+        self._cx_dag.apply_operation_back(HGate(), [qr[0]], [])
+        self._cx_dag.apply_operation_back(HGate(), [qr[1]], [])
+        self._cx_dag.apply_operation_back(CXGate(), [qr[1], qr[0]], [])
+        self._cx_dag.apply_operation_back(HGate(), [qr[0]], [])
+        self._cx_dag.apply_operation_back(HGate(), [qr[1]], [])
+
+        self._ecr_dag = DAGCircuit()
+        qr = QuantumRegister(2)
+        self._ecr_dag.add_qreg(qr)
+        self._ecr_dag.apply_operation_back(RYGate(-pi / 2), [qr[0]], [])
+        self._ecr_dag.apply_operation_back(RYGate(pi / 2), [qr[1]], [])
+        self._ecr_dag.apply_operation_back(ECRGate(), [qr[1], qr[0]], [])
+        self._ecr_dag.apply_operation_back(HGate(), [qr[0]], [])
+        self._ecr_dag.apply_operation_back(HGate(), [qr[1]], [])
+
+    def _rzx_dag(self, theta):
+
+        rzx_dag = DAGCircuit()
+        qr = QuantumRegister(2)
+        rzx_dag.add_qreg(qr)
+        rzx_dag.apply_operation_back(HGate(), [qr[0]], [])
+        rzx_dag.apply_operation_back(HGate(), [qr[1]], [])
+        rzx_dag.apply_operation_back(RZXGate(theta), [qr[1], qr[0]], [])
+        rzx_dag.apply_operation_back(HGate(), [qr[0]], [])
+        rzx_dag.apply_operation_back(HGate(), [qr[1]], [])
+        return rzx_dag
+
+    def is_native_cx(self, qubit_pair: Tuple):
```

Comment:
```suggestion
    def is_native_cx(self, qubit_pair: Tuple) -> bool:
```
please make sure to add type hints.